tests/rados/test_election_strategies.py

```diff
@@ -222,3 +283,382 @@ def check_mon_status(rados_obj, mon_obj):
             return False
         log.info(f"Mon {mon} is in healthy state: {status} | {desc}")
     return True
+
+
+def add_all_mons_to_disallow_list(mon_obj) -> None:
+    """Method to validate last Mon cannot be added to disallowed leaders list
+
+    Steps:
+    - Get list of all Mons from the cluster
+    - Add all Mons to disallowed leaders list
+    - Validate last allowed Mon cannot be added to disallow list
+    - Revert cluster back to original state by removing all Mons from disallow list
+
+    Args:
+        mon_obj: object MonElectionStrategies
+    Returns:
+        Raises Exception when test fails
+    Example:
+
+    """
+    log.info(
+        "\n -------------------------------------------"
+        "\n Starting validation:"
+        "\n `last allowed Mon` cannot be added to disallow list"
+        "\n -------------------------------------------"
+    )
+
+    mons = list(mon_obj.get_mon_quorum().keys())
+    last_allowed_mon = mons.pop()
+
+    for mon in mons:
+        log.info(f"Adding mon {mon} to disallow list")
+        if not mon_obj.set_disallow_mon(mon=mon):
+            raise AssertionError(f"could not add Mon {mon} to disallowed list")
+        log.info(f"Successfully added mon {mon} to disallow list")
+
+    log.info(
+        f"Adding last remaining allowed mon {last_allowed_mon} to disallow list\n"
+        "Proceeding to validate error generation when last allowed mon is added to disallow list"
+    )
+    try:
+        if mon_obj.set_disallow_mon(mon=last_allowed_mon):
+            raise AssertionError(
+                f"Last allowed mon {last_allowed_mon} added to disallow list"
+            )
+
+    except AssertionError as ae:
+        raise AssertionError(ae.args[0])
+
+    except Exception as e:
+        expected_error_msg = f"Error EINVAL: mon.{last_allowed_mon} is the only remaining allowed leader!"
+        generated_error_msg = e.args[0].split("\n")[4]
+        if generated_error_msg != expected_error_msg:
```

Comment:
check the substring exists in the main string via "in".

Reply:
Replace error checks with 'in' keyword

https://github.com/red-hat-storage/cephci/blob/807667591404aafc2a68a4f49acd30939d61f065/tests/rados/test_election_strategies.py#L139-L150